test/e2e/nsx_inventory_test.go

```diff
@@ -0,0 +1,324 @@
+package e2e
+
+import (
+	"context"
+	"fmt"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	corev1 "k8s.io/api/core/v1"
+	networkingv1 "k8s.io/api/networking/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+func TestInventorySync(t *testing.T) {
+	t.Run("testNamespaceSync", func(t *testing.T) { testNamespaceSync(t) })
+	t.Run("testPodSync", func(t *testing.T) { testPodSync(t) })
+	t.Run("testServiceSync", func(t *testing.T) { testServiceSync(t) })
+	t.Run("testNodeSync", func(t *testing.T) { testNodeSync(t) })
+	t.Run("testNetworkPolicySync", func(t *testing.T) { testNetworkPolicySync(t) })
+	t.Run("testIngressSync", func(t *testing.T) { testIngressSync(t) })
+}
+
+// testNamespaceSync tests that a Kubernetes namespace is synced to the NSX inventory as a ContainerProject
+func testNamespaceSync(t *testing.T) {
+	_, deadlineCancel := context.WithTimeout(context.Background(), defaultTimeout)
+	defer deadlineCancel()
+
```

Comment:
Why we have these in each function? It seems the deadline context is never used?